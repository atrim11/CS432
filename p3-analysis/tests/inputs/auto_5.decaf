def int getMax(int a, int b) {
    if (a > b) {
        return a;
    } else {
        return b;
    }
}

def void printMax(int x) {
    print_str(x);  // Error: should use print_int, not print_str
}

int values[4];
def int main() {
    // Incorrect return type
    int max;
    int a;
    int a;
    int sum;
    int a;
    max = printMax(10);  // Error: void function used in assignment

    // Array access issues
    values[5] = 20;  // Error: index out of bounds

    // Parameter type mismatch
    result = getMax("string", 5);  // Error: first parameter should be int

    // Valid function calls
    max = getMax(10, 20);
    printMax(max);

    // Invalid conditional expression
    if (max + 10) {  // Error: non-boolean expression in 'if'
        print_str("Condition error");
    }

    // Variable redeclaration
    a = 5;
    a = 10;  // Error: redeclaration of 'a'

    // Using undeclared variables
    sum = a + b;  // Error: 'b' is undeclared

    text = badReturn();  // Error propagating from incorrect return type
}
