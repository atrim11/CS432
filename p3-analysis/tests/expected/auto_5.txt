Type mismatch in parameter 0 of call to 'print_str': expected str but found int on line 10
Type mismatch: int is incompatible with void on line 21
Symbol 'result' undefined on line 27
Type mismatch in parameter 0 of call to 'getMax': expected int but found str on line 27
Type mismatch: bool expected but int found on line 34
Symbol 'b' undefined on line 43
Symbol 'text' undefined on line 45
Symbol 'badReturn' undefined on line 45
Duplicate symbols named 'a' in scope started on line 14
